### PR TITLE
fix: making api tokens creation work

### LINKED ISSUE #19994: API Token creation not working (due to token-duration)
## Bug report
API Token creation is not working at the moment.

### Required System information

- Node.js version: v18.17.0
- Strapi version: 5.0.0-beta.1
- Database: sqllite
- Is your project Javascript or Typescript: JS
Environment: development
OS: darwin-arm64
Node/Yarn Version: yarn/1.22.19 npm/? node/v18.17.0 darwin arm64
Edition: Community

### Describe the bug

In the process of testing the v5-beta within my given architecture, I was not able to create an API token due to errors in the token-duration setting field. No matter what I choose from the dropdown, I'll always run into getting an error like: "lifespan must be one of the following values: , 604800000, 2592000000, 7776000000". I can't set this manually and I can't proceed to create the token in any form.

### Steps to reproduce the behavior

1. Install the beta with npx create-strapi-app@beta beta --quickstart
2. Create admin user and proceed to the "General Settings > API Token" section
3. Try to create a token with ANY token type or token duration and you'll run into an error in the token-duration field

### Expected behavior

Creation of the Token with correct duration settings

### Screenshots

screenshot 2024-04-02 at 17 17 30

### PR DESCRIPTION
**Pull request by Bassel17**
### What does it do?

- Makes sure that the lifespan is a number not a string
- Makes sure the user gets navigated to the correct page after successful creation
- Add e2e tests to for creating API tokens

### Why is it needed?

- Creating API tokens was broken

### How to test it?

- Attempt to create an API token, it should work and you should be able to copy the token

### Related issue(s)/PR(s)

fix #19994

### CODE REVIEW

File: packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx [hunk 1]
```diff
@@ -190,7 +190,8 @@ export const EditView = () => {
         const res = await createToken({
           ...body,
           // lifespan must be "null" for unlimited (0 would mean instantly expired and isn't accepted)
-          lifespan: body?.lifespan || null,
+          lifespan:
```
Comment (innerdvations) [inline packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:193]: Just to confirm, the original PR updated both API and Transfer tokens to be more similar (I think 6 file changes in total). Did you check if we lost any of the other changes in the merge that might need to be copied over too?

Comment (Bassel17) [reply packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:193]: Nope we didn't lose any other changes, the server changes weren't lost

File: packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx [hunk 2]
```diff
@@ -190,7 +190,8 @@ export const EditView = () => {
...
+            body?.lifespan && body.lifespan !== '0' ? parseInt(body.lifespan.toString(), 10) : null,
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:194]: why are you converting it to a string to then parse it back to a number? Is it not already a number prior?

Comment (innerdvations) [reply packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:194]: You can find the reasoning [here](https://github.com/strapi/strapi/pull/19621)

Comment (joshuaellis) [reply packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:194]: that is so confusing

Comment (innerdvations) [reply packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx:194]: I just added one

**Review by innerdvations — APPROVED** [review]
LGTM, also QAd it.

After creating an unlimited token, both this and transfer tokens display an empty select item instead of showing "Unlimited"

Is that something you can quickly solve here, or do you want to open another issue for it?

**Pull request merged**